src/devices/StepperMotor/StepperMotor.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Device.Gpio;
+using System.Diagnostics;
+
+namespace Iot.Device.StepperMotor
+{
+    /// <summary>
+    /// This class is for controlling stepper motors that are controlled by a 4 pin controller board.
+    /// </summary>
+    /// <remarks>It is tested and developed using the 28BYJ-48 stepper motor and the ULN2003 driver board.</remarks>
+    public class StepperMotor : IDisposable
+    {
+        private int _pin1;
+        private int _pin2;
+        private int _pin3;
+        private int _pin4;
+        private int _steps = 0;
+        private int _engineStep = 0;
+        private int _currentStep = 0;
+        private StepperMode _mode;
+        private bool _isClockwise = true;
+        private GpioController _controller;
+        private Stopwatch _stopwatch;
+        private uint _rpm;
+        private long stepMicrosecondsDelay;
+
+        /// <summary>
+        /// Initialize a StepperMotor class.
+        /// </summary>
+        /// <param name="pin1">The GPIO pin number which corresponds pin A on ULN2003 driver board.</param>
+        /// <param name="pin2">The GPIO pin number which corresponds pin B on ULN2003 driver board.</param>
+        /// <param name="pin3">The GPIO pin number which corresponds pin C on ULN2003 driver board.</param>
+        /// <param name="pin4">The GPIO pin number which corresponds pin D on ULN2003 driver board.</param>
+        /// <param name="mode">The stepper's mode.</param>
+        /// <param name="rpm">Revolutions per minute.</param>
+        public StepperMotor(int pin1, int pin2, int pin3, int pin4, StepperMode mode = StepperMode.HalfStep, uint rpm = 15)
```

Comment:
I think rpm should be signed number. `rpm = 0` should be no movement, `rpm < 0`, rotation one way, `rpm > 0` rotating the other way

Reply:
Hmm I bit doubt about using negative RPM. We already use negative steps for rotate counterclockwise.

                    motor.Mode = StepperMode.HalfStep;
                    motor.RPM = 8;
                    motor.Step(-2048);

Will the following code be misleading If I set RPM as -8?